File: go/gin/internal/modules/branch/module.go
package branch

import (
	auth "tcc/internal/modules/auth/middleware"

	"github.com/gin-gonic/gin"
)

func BranchModule(router *gin.Engine) {
	service := BranchService{
		repository: BranchRepository{},
	}
	controller := BranchController{
		service: service,
	}

	group := router.Group("/agencias")
	{
		group.GET("", auth.AuthMiddleware(), controller.FindAll)
	}
}

File: rust/actix/src/modules/branch/module.rs
use actix_web::{web, Scope, dev::ServiceFactory};
use actix_web::dev::ServiceRequest;
use actix_web::dev::ServiceResponse;
use actix_web::Error;
use crate::modules::auth::middleware::AuthMiddleware;
use super::controller::BranchController;

pub fn branch_module() -> Scope<impl ServiceFactory<ServiceRequest, Config = (), Response = ServiceResponse, Error = Error, InitError = ()>> {
    web::scope("/agencias")
        .wrap(AuthMiddleware)
        .route("", web::get().to(BranchController::find_all))
}
